Simplify implementation of '<canvas>' 'width' and 'height' attributes.

eefriedman opened the pull request:
Strictly speaking, this affects correctness for extremely large width and height values... but that's unlikely to matter in practice.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/servo/servo/8243)

<!-- Reviewable:end -->

eefriedman commented:
Err, this isn't right; it needs a `parse_plain_attribute` implementation.  Sorry about the noise.

eefriedman commented:
Should be fixed now.

Ms2ger commented:
-S-awaiting-review +S-needs-code-changes 
<!-- Reviewable comment -K1mp6hrHtBd7RUb6OZz:1458266033 -->

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

<sup>**[components/script/dom/htmlcanvaselement.rs, line 126 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8243#-K1mohagguOoEj73vT6-)** ([raw file](https://github.com/servo/servo/blob/6e332dbc84acc53aecdf98d81cb220acc9a86821/components/script/dom/htmlcanvaselement.rs#L126)):</sup>
This will try to create roots from the layout thread.
<!-- Reviewable comment -K1mohahWbLs1qhiyLtG:-1012483121 -->

---

<sup>**[components/script/dom/htmlcanvaselement.rs, line 291 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8243#-K1mp1r_QHyC_aVxBIRP)** ([raw file](https://github.com/servo/servo/blob/6e332dbc84acc53aecdf98d81cb220acc9a86821/components/script/dom/htmlcanvaselement.rs#L291)):</sup>
No need for the `recreate` local.
<!-- Reviewable comment -K1mp1r_QHyC_aVxBIRQ:-2120484 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8243)
<!-- Sent from Reviewable.io -->

Ms2ger commented:
Btw, I checked... This does fail all the tests.

eefriedman committed:
Simplify implementation of '<canvas>' 'width' and 'height' attributes.

eefriedman commented:
Addressed review comments.  (And I'll try to be more careful about running tests before posting a patch in the future.)
<!-- Reviewable comment -K1oyPvQwGEDCDz71pw1:-944987982 -->

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8243)
<!-- Sent from Reviewable.io -->

frewsxcv commented:
<sup>**[components/script/dom/htmlcanvaselement.rs, line 129 \[r2\]](https://reviewable.io:443/reviews/servo/servo/8243#-K1qk3wnCYnH6FcXeBbw)** ([raw file](https://github.com/servo/servo/blob/6a7e6e78da16e30e81aa22b7d1b43d9a3bbbb89b/components/script/dom/htmlcanvaselement.rs#L129)):</sup>
Not a big deal, but this could be `map_or` instead of `map..unwrap_or`
<!-- Reviewable comment -K1qk3wnCYnH6FcXeBbx:483505678 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8243)
<!-- Sent from Reviewable.io -->

eefriedman commented:
Updated.

eefriedman commented:
@Ms2ger Ping.

Ms2ger commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit 4fc0a66 has been approved by `Ms2ger`

<!-- @bors-servo r=Ms2ger 4fc0a66ba1cc9bd45673c7d2095ca34b90e09197 -->

bors-servo commented:
:hourglass: Testing commit 4fc0a66 with merge 46ad8b9...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/251)

jdm commented:
@bors-servo: retry
* #8311

bors-servo commented:
:hourglass: Testing commit 4fc0a66 with merge b39035c...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/242)

frewsxcv commented:
@bors-servo retry

#7791

bors-servo commented:
:zap: Previous build results for [android](http://build.servo.org/builders/android/builds/231), [gonk](http://build.servo.org/builders/gonk/builds/239), [linux-dev](http://build.servo.org/builders/linux-dev/builds/235), [linux-rel](http://build.servo.org/builders/linux-rel/builds/254), [mac-dev-ref-unit](http://build.servo.org/builders/mac-dev-ref-unit/builds/247), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/257) are reusable. Rebuilding only [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/242)...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/231), [gonk](http://build.servo.org/builders/gonk/builds/239), [linux-dev](http://build.servo.org/builders/linux-dev/builds/235), [linux-rel](http://build.servo.org/builders/linux-rel/builds/254), [mac-dev-ref-unit](http://build.servo.org/builders/mac-dev-ref-unit/builds/247), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/245), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/257)
